homeassistant/components/tasmota/mixins.py

```diff
@@ -0,0 +1,124 @@
+"""Tasnmota entity mixins."""
+import logging
+
+from homeassistant.components.mqtt.const import MQTT_CONNECTED, MQTT_DISCONNECTED
+from homeassistant.components.tasmota.discovery import set_discovery_hash
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+
+from .discovery import TASMOTA_DISCOVERY_ENTITY_UPDATED, clear_discovery_hash
+
+DATA_MQTT = "mqtt"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TasmotaEntity(Entity):
+    """Base class for Tasmota entities."""
+
+    def __init__(self, tasmota_entity) -> None:
+        """Initialize."""
+        self._tasmota_entity = tasmota_entity
+
+
+class TasmotaAvailability(TasmotaEntity):
+    """Mixin used for platforms that report availability."""
+
+    def __init__(self, **kwds) -> None:
+        """Initialize the availability mixin."""
+        self._available = False
+        super().__init__(**kwds)
+
+    async def async_added_to_hass(self) -> None:
+        """Subscribe MQTT events."""
+        await super().async_added_to_hass()
+        self._tasmota_entity.set_on_availability_callback(self.availability_updated)
+        self.async_on_remove(
+            async_dispatcher_connect(self.hass, MQTT_CONNECTED, self.async_mqtt_connect)
```

Comment:
The signal is an mqtt integration detail. If we want to make this a public interface the mqtt integration should make a helper function to listen to the mqtt client connected event for interested callers.

Reply:
Fixed.